docs/sources/developers/plugins/extend-the-grafana-ui-with-links.md

```diff
@@ -1,25 +1,78 @@
 ---
-title: Use extensions to add links to app plugins
+title: Use extensions to add new functionality in the Grafana user interface
 description: Learn how to add links to the Grafana user interface from an app plugin
 ---
 
-# Use extensions to add links to app plugins
+# The when, why and how of UI extensions
 
-You can use the Plugin extensions API with your Grafana app plugins to add links to the Grafana UI. This feature lets you send users to your plugin's pages from other spots in the Grafana application.
+Before we go into details we need to cover the two major concepts in the UI extensions feature of Grafana.
+
+1. Extension point - a place in the UI where plugins can provide new functionality to the end user. Examples of extension points are the dashboard panel menu or toolbar in explore.
+2. Extensions - New functionality, usually registered by an app plugin, that will be displayed at an extension point. Examples of extensions are the possibility to create an incident directly from the dashboard panel menu.
```

Comment:
nit
```suggestion
2. Extensions - new functionality, usually registered by an app plugin, that will be displayed at an extension point. Examples of extensions are the possibility to create an incident directly from the dashboard panel menu.
```

However, I do think we need to be more explicit on what functionality can be added, or expand the examples in a more generic way i.e. Link/Modal/Cmd etc

Reply:
i.e. provide a link to navigate a user to a view within a separate plugin, or open a modal menu to allow the user to trigger functionality from a separate plugin within their current context

Reply:
- You can link the user to a different view within the extending plugin.
- Open a different workflow in a modal that sits on top of the current view.
- Trigger background tasks.